### ISSUE TITLE
Merge room directory and start chat

### ISSUE THREAD

**Issue by jo-so** [vector-im/riot-web#7715]
I've a suggestion. How about merging the *room directory* and *start chat* as entry point for new conversations? It would be nice if this starting point would support something to open matrix.to URLs. I would like to paste URLs like `https://matrix.to/#/!TTajZMFjIFlHBynrvu:yuhu.ddns.net/$1542524557556iSLhn:alea.gnuu.de` into a box and Riot does the right thing, e.g. it recognizes that this is a link to a message or `https://riot.im/app/#/room/#riot:matrix.org` is a link to a room and so on.

**Comment by lampholder** [comment]
Responding to the two separate parts of this:

- Merge _room directory_ and _start chat_
    This might be nice, but I think would be blocked on the room directory's being much more performant and user-friendly.
- Somewhere to paste URLs
    This would be _super_ useful for the electron app. It would need some UX design thought, though, so perhaps this could be implemented as `/command` first?

**Comment by turt2live** [comment]
> though, so perhaps this could be implemented as /command first?

I snuck this in to the last release: can type `/goto <matrix.to url>` and it'll work.

**Comment by jo-so** [comment]
Travis Ralston hat am Fr 23. Nov, 15:14 (+0000) geschrieben:
> > though, so perhaps this could be implemented as /command first?
> 
> I snuck this in to the last release: can type `/goto <matrix.to url>` and it'll work.

Very cool!

I've tried `/goto https://riot.im/app/#/room/#riot:matrix.org`, but it
gives me `Command error Usage: /join <room-alias>`.

Do you think *goto* could also jump to a room? `/goto riot` could to the
same like

1. Ctrl-k (filter rooms)
1. enter `riot`
1. key down (select the first match)
1. enter

This would be very handy, because you don't have to leave the input field
for changing a room.

Maybe `#riot` and `@user` could narrow the filter to *rooms* and *people*.

-- 
Die zehn Gebote Gottes enthalten 172 Wörter, die amerikanische
Unabhängigkeitserklärung 300 Wörter, die Verordnung der europäischen
Gemeinschaft über den Import von Karamelbonbons exakt 25911 Wörter.

**Comment by r4dh4l** [comment]
**Room creation**

I would like to suggest to even merge "room directory", "start chat" and "create room". The current UI design seems to unnecessarily hide the "everything is a room" philosophy of Matrix. I think it would much better to just offer a `+` symbol instead of 3 buttons. The `+` then offers:

- create a room just for me (room symbol could be my avatar)
- create a room for me and another user (with a second question, if the creator or both users should be admins; room symbol could be a combination of both user avatars, maybe two avatar circles where my own one is pointing to the addressed one: `o -> o`)
- create a room for me and multiple users (room symbol could be a circle of all avatar circles in the room. like "pearl necklace" - or a custom avatar like now)
- join an existing room (offering to inupt the room ID or to choose from a list of public rooms)

There should be a hint like "In Matrix everything is managed in rooms. You can have a room just for yourself, for you and another person and for a group of people - flexible to adjust the amount of people any time.

This would solve the problem that a "Start chat" currently confuses all people I introduced into Matrix/Riot because you can invite more than one person in the end of the dialogue. I know that the development of Riot wanted to use structures the majority of people know from common instant messengers but it is better to explain new users one time that "everything is a room" so that it makes not much sense to distinguish between single and group chat rooms.

**Notication indicator**

Another idea just came up: Please add a small symbol to the room avatar indicating the notification level.

**Room categories/filters**

Furthermore the sections "People", "Rooms" and "Low Priority" make not much sense because as far as I watched people using Riot they accidently move "rooms" to "people" and "people" to "rooms". Same with "low priority" which has, as far as I recognized, has no effect on the notification settings and is just another category.
The categories work when you have only a small amount of contacts but with 30+ contacts you have to use the "filter room names" so why using filters instead in the first place?

- show all rooms created by me
- show all rooms with me as only participant
- show all rooms with me as only participant and bots
- show all rooms with just me an another participant
- show all rooms with multiple participants
- show all rooms marked as favorite
- show all rooms marked as "low priority"
- show all rooms with notification level "all messages (noisy)"
- show all rooms with notification level "all messages"
- show all rooms with notification level "mute"
- show all rooms I left (historical)

**Comment by aaronraimist** [comment]
> about merging the _room directory_ and _start chat_ as entry point for new conversations?

this is pretty much how it works in 1.0.0 although not with start chat, only with the + button next to the general rooms list which I think makes more sense even though yes "everything is a room".

**Comment by turt2live** [comment]
I think everything raised here has been addressed - if there's something missing, please open new issues for each distinct part.

**Comment by r4dh4l** [comment]
> I think everything raised here has been addressed - if there's something missing, please open new issues for each distinct part.

Well, not really but as suggested I started a new issue: https://github.com/vector-im/riot-web/issues/8771
